src/Search/Filter/index.js

```diff
@@ -1,52 +1,223 @@
 import React from 'react';
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
+import { map, includes } from 'lodash';
+import pluralize from 'pluralize-ru';
 import Filter from './Filter';
 import FlyDates from './FlyDates';
 import FlyDuration from './FlyDuration';
 import Reset from './Reset';
-import Transfers from './Transfers';
-import Airlines from './Airlines';
+import Checkbox from './Checkbox';
+import { translate } from '../../translate';
 import { transfers, flyDates, flyDurations, airlines } from './filtersData';
 
 const Wrapper = styled.div`
   background: #fff;
   border-radius: 4px;
 `;
 
-const Filters = (props) => {
-  const { origin, destination } = props;
-
-  return (
-    <Wrapper>
-      <Transfers transfers={transfers} />
-      <Filter label="ВРЕМЯ ВЫЛЕТА И ПРИБЫТИЯ" isOpened>
-        <FlyDates ranges={flyDates[0].ranges} origin={origin} destination={destination} />
-        <FlyDates ranges={flyDates[1].ranges} origin={destination} destination={origin} />
-      </Filter>
-      <Filter label="Багаж" />
-      <Filter label="Длительность пересадки" />
-      <Filter label="Время в пути" isOpened>
-        <FlyDuration ranges={flyDurations[0].ranges} origin={origin} destination={destination} />
-        <FlyDuration ranges={flyDurations[1].ranges} origin={destination} destination={origin} />
-      </Filter>
-      <Airlines airlines={airlines} />
-      <Filter label="Аэропорты" />
-      <Filter label="Аэропорт пересадки" count={71} />
-      <Filter label="Агенства" count={26} />
-      <Reset />
-    </Wrapper>
-  );
+const GroupTitle = styled.h4`
+  font-size: 0.75rem;
+  margin: 20px 16px 10px 16px;
+`;
+
+const defaultListState = {
+  transfers: map(transfers.list, 'id'),
+  alliances: map(airlines.alliances.list, 'id'),
+  aircompanies: map(airlines.aircompanies.list, 'id'),
 };
 
+class Filters extends React.Component {
+  state = {
+    transfers: {
+      all: true,
+      list: defaultListState.transfers,
+    },
+    alliances: {
+      all: true,
+      list: defaultListState.alliances,
+    },
+    aircompanies: {
+      all: true,
+      list: defaultListState.aircompanies,
+    },
+  };
+
+  setCheckboxes = (id, group) => {
+    const list = includes(this.state[group].list, id)
+      ? this.state[group].list.filter(value => value !== id)
+      : [...this.state[group].list, id];
+
+    this.setState({
+      [group]: {
+        all: list.length === defaultListState[group].length,
+        list,
+      },
+    });
+  };
+
+  setCheckboxesAll = (id, group) => {
+    const allChecked = !this.state[group].all;
```

Comment:
https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b